apps/antalmanac/src/components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog.tsx

```diff
@@ -43,188 +36,214 @@ export interface RepeatingCustomEvent {
+        setTitle(defaultCustomEventValues.title);
+        setDays(defaultCustomEventValues.days);
     };
 
-    handleAddToCalendar = () => {
-        if (!this.state.days.some((day) => day) || this.state.scheduleIndices.length === 0) return;
+    const handleAddToCalendar = () => {
+        if (disabled) return;
 
         const newCustomEvent = {
-            color: this.props.customEvent ? this.props.customEvent.color : '#551a8b',
-            title: this.state.title,
-            days: this.state.days,
-            start: this.state.start,
-            end: this.state.end,
-            customEventID: this.props.customEvent ? this.props.customEvent.customEventID : Date.now(),
+            color: props.customEvent ? props.customEvent.color : '#551a8b',
+            title: title,
+            days: days,
+            start: start,
+            end: end,
+            customEventID: props.customEvent ? props.customEvent.customEventID : Date.now(),
         };
 
-        if (this.props.customEvent) editCustomEvent(newCustomEvent, this.state.scheduleIndices);
-        else addCustomEvent(newCustomEvent, this.state.scheduleIndices);
-    };
-
-    handleSelectScheduleIndices = (scheduleIndices: number[]) => {
-        this.setState({ scheduleIndices: scheduleIndices });
-    };
+        resetForm();
 
-    isAddDisabled = () => {
-        return !(this.state.scheduleIndices.length && this.state.days.some(Boolean));
+        props.customEvent
+            ? editCustomEvent(newCustomEvent, scheduleIndices)
+            : addCustomEvent(newCustomEvent, scheduleIndices);
     };
 
-    render() {
-        return (
-            <>
-                {this.props.customEvent !== undefined ? (
-                    // Dumb ternary below added to get rid of TypeScript possibly undefined compile error
-                    <IconButton
-                        onClick={() =>
-                            this.setState({
-                                open: true,
-                                scheduleIndices: AppStore.schedule.getIndexesOfCustomEvent(
-                                    this.props.customEvent ? this.props.customEvent.customEventID : 0
-                                ),
-                            })
-                        }
+    const handleEventNameChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = useCallback(
+        (event) => {
+            setTitle(event.target.value);
+        },
+        [setTitle]
+    );
+
+    const handleStartTimeChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = useCallback(
+        (event) => {
+            setStart(event.target.value);
+        },
+        [setStart]
+    );
+
+    const handleEndTimeChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = useCallback(
+        (event) => {
+            setEnd(event.target.value);
+        },
+        [setEnd]
+    );
+
+    const handleDayChange = useCallback(
```

Comment:
Ok, these are way too many callbacks. Do you wanna plug in react-hook-form for this yourself? Or would you want me to take care of it?

Reply:
_To be implemented in the future_